### PR TITLE
Bluetooth: Att: Add ATT min/max handle defines

### PR DESCRIPTION
**Pull request by Thalley**
Adds #defines for the minimum and maximum attribute
handles in ATT. These are useful for when setting e.g.
the start and end handle when doing a full GATT discover.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

File: include/bluetooth/att.h [hunk 1]
```diff
@@ -47,6 +47,10 @@ extern "C" {
 /* Version 5.2, Vol 3, Part F, 3.2.9 defines maximum attribute length to 512 */
 #define BT_ATT_MAX_ATTRIBUTE_LEN		512
 
+/* Handle 0x0000 is reserved for future use */
+#define BT_ATT_MIN_ATTTRIBUTE_HANDLE		0x0001
+#define BT_ATT_MAX_ATTTRIBUTE_HANDLE		0xffff
```
Comment (Thalley) [inline include/bluetooth/att.h:52]: I also considered `BT_ATT_FIRST_ATTTRIBUTE_HANDLE` and `BT_ATT_LAST_ATTTRIBUTE_HANDLE` as well as just having these as 
`BT_ATT_MIN_HANDLE` (i.e. without `ATTRIBUTE`). 

Please let me know if you reviewers have any preferences.

Comment (Thalley) [reply include/bluetooth/att.h:52]: Might add that the reason for using `MAX` is that the spec states:
> An attribute handle of value 0xFFFF is known as the maximum attribute handle.

Comment (jhedberg) [reply include/bluetooth/att.h:52]: I'm fine with the current naming as it's consistent with the existing `BT_ATT_MAX_ATTRIBUTE_LEN`, but please don't do camel-case `Att` in the commit message but capitalize it all :)

Comment (joerchan) [reply include/bluetooth/att.h:52]: Setting start handle to first handle and end handle to last handle makes more sense to me.

We can put the statement in the comment for clarification though.

Comment (Thalley) [reply include/bluetooth/att.h:52]: @joerchan So what do you prefer? Having the define as FIRST/LAST and then have MIN/MAX in the comments, or vice versa?

Comment (joerchan) [reply include/bluetooth/att.h:52]: FIRST/LAST in the name, and that spec line in the comment above the last.

Comment (Thalley) [reply include/bluetooth/att.h]: @joerchan Updated

File: include/bluetooth/att.h [hunk 2]
```diff
@@ -47,6 +47,10 @@ extern "C" {
 /* Version 5.2, Vol 3, Part F, 3.2.9 defines maximum attribute length to 512 */
 #define BT_ATT_MAX_ATTRIBUTE_LEN		512
 
+/* Handle 0x0000 is reserved for future use */
+#define BT_ATT_FIRST_ATTTRIBUTE_HANDLE		0x0001
```
Comment (Thalley) [inline include/bluetooth/att.h:51]: Just FYI, the indentation is right, this is just Github that is bad a showing tabs properly :)

Comment (joerchan) [reply include/bluetooth/att.h:51]: Should use spaces for alignment though.

Comment (Thalley) [reply include/bluetooth/att.h:51]: Doesn't the style guide say to use whatever the file currently uses? 

If switching to spaces, it should be done for all the #defines, and is out of scope of this PR imo.

Comment (joerchan) [reply include/bluetooth/att.h:51]: Does it? I'm not sure, but I use spaces everywhere. I think most files has a mix of both at this point.
No big deal though.

Comment (Thalley) [reply include/bluetooth/att.h]: Hmm, looks like it's only for certain things that you should adopt what's already used in the code. Updated to use spaces.

**Review by joerchan — APPROVED** [review]

File: samples/bluetooth/central_hr/src/main.c [hunk 3]
```diff
@@ -205,8 +205,8 @@ static void connected(struct bt_conn *conn, uint8_t conn_err)
 		memcpy(&uuid, BT_UUID_HRS, sizeof(uuid));
 		discover_params.uuid = &uuid.uuid;
 		discover_params.func = discover_func;
-		discover_params.start_handle = 0x0001;
-		discover_params.end_handle = 0xffff;
+		discover_params.start_handle = BT_ATT_FIRST_ATTTRIBUTE_HANDLE;
+		discover_params.end_handle = BT_ATT_LAST_ATTTRIBUTE_HANDLE;
```
Comment (Vudentz) [inline samples/bluetooth/central_hr/src/main.c:209]: We can possible make it even cleaner with use of a macro the init the struct with these values.

**Review by Vudentz — APPROVED** [review]
LGTM.

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
@Thalley please update the commit messages - they still contain references to the original define names (MIN/MAX)

**Comment by Thalley** [comment]
> @Thalley please update the commit messages - they still contain references to the original define names (MIN/MAX)

I knew I had forgotten something... Thanks!

**Pull request merged**